config/config/src/main/java/io/helidon/config/BuilderImpl.java

```diff
@@ -427,14 +430,13 @@ private ConfigSourcesRuntime buildConfigSources(ConfigContextImpl context) {
             envVarAliasGeneratorEnabled = true;
         }
 
-        boolean nothingConfigured = sources.isEmpty();
+        boolean nothingConfigured = sources.isEmpty() && !sourcesConfigured;
 
         if (nothingConfigured) {
             // use meta configuration to load all sources
-            MetaConfig.configSources(mediaType -> context.findParser(mediaType).isPresent(), context.supportedSuffixes())
-                    .stream()
+            MetaConfigFinder.findConfigSource(mediaType -> context.findParser(mediaType).isPresent(), context.supportedSuffixes())
```

Comment:
Should this also catch & log the exception similar to `Config.Builder.metaConfig()` ?

Reply:
This is not using meta-configuration. It only looks for default config sources (such as `application.yaml`), so it should not throw that exception at all.

Reply:
If I execute this code with a bad `meta-config.yaml` in the current working directory, it throws an exception:

```java
MetaConfig.configSources(mediaType -> true, List.of("yaml"))
        .stream()
        .count();
```

```
Exception in thread "main" java.lang.IllegalArgumentException: Config source of type properties is not supported. Supported types: [file, prefixed, classpath, system-properties, directory, inlined, url, environment-variables]
	at io.helidon.config/io.helidon.config.MetaProviders.lambda$configSource$7(MetaProviders.java:116)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.helidon.config/io.helidon.config.MetaProviders.configSource(MetaProviders.java:116)
	at io.helidon.config/io.helidon.config.MetaConfig.configSource(MetaConfig.java:201)
	at io.helidon.config/io.helidon.config.MetaConfig.lambda$configSources$1(MetaConfig.java:239)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at io.helidon.config/io.helidon.config.MetaConfig.lambda$configSources$2(MetaConfig.java:239)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at io.helidon.common.mapper/io.helidon.common.mapper.OptionalValue.ifPresent(OptionalValue.java:188)
	at io.helidon.config/io.helidon.config.MetaConfig.configSources(MetaConfig.java:239)
	at java.base/java.util.Optional.map(Optional.java:260)
	at io.helidon.config/io.helidon.config.MetaConfig.configSources(MetaConfig.java:249)
```

Reply:
will check

Reply:
The code I shared above uses `MetaConfig.configSources` whereas this PR changes to ` MetaConfigFinder.findConfigSource` ; so my comment is not relevant.